File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        max_total = 0
        global_total = nums[0]
        
        for i in range(len(nums)):
            max_total = max(nums[i], nums[i] + max_total)
            
            
            if max_total > global_total:
                global_total = max_total
            
        return global_total

File: 53-maximum-subarray/53-maximum-subarray.go
func maxSubArray(nums []int) int {
    var prevSum int
    var maxSum int
    for i := range nums {
        if i == 0 {
            prevSum = nums[i]
            maxSum = nums[i]
            continue
        } 
        if prevSum + nums[i] < nums[i] {
            prevSum = nums[i]
        } else {
            prevSum += nums[i]
        }
        
        if maxSum < prevSum {
            maxSum = prevSum
        }
    }
    return maxSum
}